addons/mrp/static/src/components/bom_overview_line/mrp_bom_overview_line.js

```diff
@@ -0,0 +1,140 @@
+/** @odoo-module **/
+
+import { useService } from "@web/core/utils/hooks";
+import { _t } from "@web/core/l10n/translation";
+import { formatFloat, formatFloatTime, formatMonetary } from "@web/views/fields/formatters";
+
+const { Component } = owl;
+
+export class BomOverviewLine extends Component {
+    setup() {
+        this.actionService = useService("action");
+        this.ormService = useService("orm");
+    }
+
+    //---- Handlers ----
+
+    async goToRoute(routeType) {
+        if (routeType == "buy") {
+            return this.goToAction(this.data.link_id, this.data.link_model);
+        } else if (routeType == "manufacture") {
+            return this.goToAction(this.data.bom_id, "mrp.bom");
+        }
+    }
+
+    async goToAction(id, model) {
+        return this.actionService.doAction({
+            type: "ir.actions.act_window",
+            res_model: model,
+            res_id: id,
+            views: [[false, "form"]],
+            target: "current",
+            context: {
+                active_id: id,
+            },
+        });
+    }
+
+    async goToForecast() {
+        const action = await this.ormService.call(
+            "product.product",
+            "action_product_forecast_report",
+            [[this.data.link_id]],
+        );
+        action.context = {
+            active_model: "product.product",
+            active_id: this.data.link_id,
+        };
```

Comment:
Currently, it tries to open the forecast report for the model `product.product` but with the id of the `product.template`, which open the forecast report for the wrong product.

It should either use the `link_model` beside the `link_id` (and calling the right method, `action_product_forecast_report` or `action_product_tmpl_forecast_report`, depending of the model), either use always the `product.product` id instead of the `link_id` (I see we have a key `product`  in the `data` object, but it's a string representation of the record instead of the id)

Reply:
We also have the product_id in there ! But I guess the whole point of having this `link_model` is to use the correct action. I actually forgot I made changes to the way the `link_id` and `link_model` work, so I didn't notice it affected the forecast links. Good catch, thanks !